code/modules/mob/living/simple_animal/friendly/lizard.dm

```diff
@@ -24,13 +24,63 @@
 	can_collar = TRUE
 	mob_biotypes = MOB_ORGANIC | MOB_BEAST | MOB_REPTILE
 	gold_core_spawnable = FRIENDLY_SPAWN
+	var/eating_sound = 'sound/weapons/bite.ogg'
+	/// Lizards start with a tail
+	var/has_tail = TRUE 
+
+/mob/living/simple_animal/lizard/handle_automated_action()
+	if(!stat && !buckled)
+		if(prob(1))
+			custom_emote(EMOTE_VISIBLE, pick("sticks out its tongue", "wags its tail.", "lies down."))
+
+	if(isturf(loc) && !incapacitated())
+		for(var/mob/living/M in view(1, src))
```

Comment:
Can you cache the `view`? Looks something like this. (I didn't know what to call the var so you can use whatever)
```dm
    var/list/lizard_view = view(1, src)
	for(var/mob/living/M in lizard_view)
        ...
	for(var/obj/structure/spider/spiderling/Spider in lizard_view)
	    ...
```

Reply:
Done